lib/multiplexer/multiplexer.go

```diff
@@ -379,7 +379,7 @@ func isHTTP(in []byte) bool {
 		[]byte("PATCH"),
 	}
 	for _, verb := range methods {
-		if bytes.HasPrefix(verb, in) {
+		if bytes.HasPrefix(in, verb) {
```

Comment:
Good catch on this bug, I spotted this in my logs recently and never got around to tracking it down.

Reply:
Let's add test coverage for this bug, if we've fallen for it once then it's worth it.